.circleci/config.yml

```diff
@@ -1,27 +1,44 @@
 version: 2.1
 
+orbs:
+  buildevents: honeycombio/buildevents@0.0.2
+
 executors:
   linuxgo:
     working_directory: /go/src/github.com/honeycombio/buildevents
     docker:
       - image: circleci/golang:1.10
 
 jobs:
+  setup:
+    executor: linuxgo
+    steps:
+      - buildevents/start_trace
   test:
     executor: linuxgo
     steps:
-      - checkout
-      - run: go test -v ./...
+      - buildevents/with_job_span:
+          steps:
+            - checkout
+            ## method 1 to send a command span
+            ## buildevent/berun is a circleci friendly way to create a buildevents command span
+            - buildevents/berun:
+                bename: go_test
+                becommand: go test -v ./...
   build:
     executor: linuxgo
     steps:
-      - checkout
-      - run: go install -ldflags "-X main.Version=${CIRCLE_TAG}" ./...
-      - run: mkdir -v artifacts; cp -v $GOPATH/bin/buildevents artifacts/
-      - persist_to_workspace:
-          root: artifacts
-          paths:
-            - buildevents
+      - buildevents/with_job_span:
+          steps:
+            - checkout
+            ## method 2 to send a command span
+            ## the raw buildevents binary is also available in the $PATH but requires more arguments
+            - run: buildevents cmd $CIRCLE_WORKFLOW_ID $BUILDEVENTS_SPAN_ID go_install -- go install -ldflags "-X main.Version=${CIRCLE_TAG}" ./...
+            - run: mkdir -v artifacts; cp -v $GOPATH/bin/buildevents artifacts/
+            - persist_to_workspace:
+                root: artifacts
+                paths:
+                  - buildevents
   publish:
```

Comment:
no buildevent on publish?

Reply:
I'll add it in when doing circle API work.